src/GeneticSharp.Domain/Chromosomes/IChromosome.cs

```diff
@@ -73,11 +73,16 @@ public interface IChromosome : IComparable<IChromosome>
         Gene[] GetGenes();
 
         /// <summary>
-        /// Creates a new chromosome using the same structure of this.
+        /// Creates a new empty chromosome using the same structure of this.
         /// </summary>
         /// <returns>The new chromosome.</returns>
         IChromosome CreateNew();
 
+        /// <summary>
+        /// Initializes a chromosome with new random genes
+        /// </summary>
+        void CreateGenes();

```

Comment:
Good call, with a second thought, that was actually uncalled for. I just [commited](https://github.com/giacomelli/GeneticSharp/pull/88/commits/2bc28b995869270e381d65b6e6ff964fc5dd1800) an alternative, where the new interface method has another name and the historical protected one is left untouched with the new method rerouted to the former one.

This may seem a bit weird, and I think that this kind of change probably still justifies bumping the major version anyway, since it cTor-based initialization should be considered bad practice from now on, but at least only derived Chromosome classes implementing the Interface directly would see a breaking change, which is most probably none of them.

Anyway, I guess we could have lived without that specific change, but I figured, since this was always going to be a problem down the road, it is now a good moment to start paying the technical debt.